src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_linux.go

```diff
@@ -620,7 +620,16 @@ func savePrometheusMetrics(ctx context.Context, ps *stepParams, admin *admin.Adm
 		if err != nil {
 			return fmt.Errorf("unable to fetch metrics from the admin API: %w", err)
 		}
-		return writeFileToZip(ps, "prometheus-metrics.txt", raw)
+		err = writeFileToZip(ps, "prometheus-metrics.txt", raw)
+		if err != nil {
+			return err
+		}
+
+		pRaw, err := admin.PublicMetrics(ctx)
+		if err != nil {
+			return fmt.Errorf("unable to fetch public metrics from the admin API: %w", err)
+		}
+		return writeFileToZip(ps, "public-metrics.txt", pRaw)
```

Comment:
Should this be a separate function?

Reply:
Yeah, I'll refactor this better and have public metrics in another step :+1: